### ISSUE TITLE
Error in `glibc-2.34/elf/rtld-Rules` makefile when building `x86_64-unknown-linux-gnu`

### ISSUE THREAD

**Issue by Wazzaps** [crosstool-ng/crosstool-ng#1640]
When building the vanilla config for x86_64-unknown-linux-gnu I get:

```
[ALL  ]      make -f /b/ctng/_run_ctng@hczgqphjnjdbrzuvazqgy7lw3yedfwiw.src/.build/x86_64-unknown-linux-gnu/build/build-libc-final/multilib/elf/librtld.mk -f rtld-Rules
[ALL  ]      make[4]: Entering directory '/b/ctng/_run_ctng@hczgqphjnjdbrzuvazqgy7lw3yedfwiw.src/.build/src/glibc-2.34/elf'
[ALL  ]      rtld-Rules:40: *** This makefile is a subroutine of elf/Makefile not to be used directly.  Stop.
```

I'm following the instructions here: https://stackoverflow.com/a/52454710/2591498

And i'm running exactly:

```sh
export PATH="/usr/lib/ccache:${PATH}"
export CT_PREFIX="/bold/app/_run_ctng@hczgqphjnjdbrzuvazqgy7lw3yedfwiw"
./bootstrap 
./configure --enable-local
make -j `nproc`
./ct-ng x86_64-unknown-linux-gnu
./ct-ng menuconfig
env -u LD_LIBRARY_PATH time ./ct-ng build CT_JOBS=`nproc`
```

Should I drop ccache? Should I move to a directory without a "@" in the name?

## Info

- `crosstool-NG version 1.24.0.496_4dc9d64_dirty`
- Host OS: `Pop!_OS 21.04 (like Ubuntu 21.04)`
  - `Linux 5.13.0-7620-generic #20~1634827117~21.04~874b071-Ubuntu SMP Fri Oct 29 15:06:55 UTC  x86_64 x86_64 x86_64 GNU/Linux`
  - `gcc (Ubuntu 10.3.0-1ubuntu1) 10.3.0`
- .config and build.log: [info.zip](https://github.com/crosstool-ng/crosstool-ng/files/7611730/info.zip)

**Comment by Wazzaps** [comment]
## Update

I dropped both ccache and the "@", it failed to build cross gdb because of a python problem (which I fixed by specifying the python interpreter path to `/usr/bin/python3`), and then it build successfully.

I then did the same but in a folder with "@" in the name, which reproduced the same issue. It seems that make invocation doesn't work with a "@" in the path of the makefile.

TL;DR: Seems like "@" in the path of the build directory (not `CT_PREFIX`, that's fine) trips up the build system.

**Comment by cpackham** [comment]
That's for debugging the issue down to figuring out it was the path. I'm not entirely sure if it's a problem with ct-ng or with glibc. If you've got bandwidth for some more debugging trying the "odd" path with glibc's build tool https://sourceware.org/git?p=glibc.git;a=blob_plain;f=scripts/build-many-glibcs.py

**Comment by tbahls** [comment]
I see this same exact error compiling OpenWrt for an aarch64 target, using glibc 2.32.

In our case it was the Jenkins automated build tool that was using a path with "@" (at) in it.
